### PR TITLE
Swich to new gcloud API for GCE MIGs

### LINKED ISSUE #11686: Migrate to the new api for GCE managed instance groups
We should migrate our scripts to use the new api for managed-instance-groups in GCE:
- instead of using: "gcloud preview managed-instance-groups ..."
- use: "gcloud (beta) compute instance-groups managed ..."

 The new api is richer and will (hopefully) also resolve some issues that we have when creating a cluster, - running "gcloud preview managed-instance-groups ... create":
https://github.com/GoogleCloudPlatform/kubernetes/blob/master/cluster/gce/util.sh#L698
 doesn't guarantee that you can list its instances (this can cause errors which we actually observe)

 In the new api we can do:
gcloud (beta) compute instance-groups managed create ...
gcloud (beta) compute instance-groups managed wait-until-stable ...

cc @davidopp @mbforbes @roberthbailey

### PR DESCRIPTION
**Pull request by wojtek-t**
Fixes #11686

cc @davidopp @mbforbes @roberthbailey

### CODE REVIEW

File: cluster/gce/util.sh [hunk 1]
```diff
@@ -695,16 +677,14 @@ function kube-up {
...
       --project "${PROJECT}" \
+      --zone "${ZONE}" \
       --base-instance-name "${NODE_INSTANCE_PREFIX}" \
       --size "${NUM_MINIONS}" \
       --template "${NODE_INSTANCE_PREFIX}-template" || true;
-  # TODO: this should be true when the above create managed-instance-group
-  # command returns, but currently it returns before the instances come up due
-  # to gcloud's deficiency.
-  wait-for-minions-to-run
+  gcloud beta compute instance-groups managed wait-until-stable "${NODE_INSTANCE_PREFIX}-group"
```
Comment (roberthbailey) [inline cluster/gce/util.sh]: does this need `|| true` at the end in case of errors? or some other error checking?

**Comment by roberthbailey** [comment]
@wojtek-t why did you mark this as do not merge?

**Comment by wojtek-t** [comment]
@roberthbailey - the new API is not yet released (should be today according to MIG team from Warsaw) - this is why Jenkins tests failed. [I've tested by updating my gcloud to (still) experimental version]

**Comment by wojtek-t** [comment]
[by new API I mean support for it in gcloud]

File: cluster/gce/util.sh (hunk 1 above)
Comment (wojtek-t) [inline cluster/gce/util.sh]: Done.

**Comment by roberthbailey** [comment]
Ah. I know that the new raw API is out already; I didn't realize gcloud support hadn't rolled out yet.

**Comment by wojtek-t** [comment]
Jenkins is now failing with the following error:

    You do not currently have this command group installed.  Using it requires the installation of components: [beta]
    ERROR: (gcloud) You cannot perform this action because you do not have permission to modify the Google Cloud SDK installation directory [/usr/local/share/google/google-cloud-sdk].

How can we workaround it (and do we want?)?

**Comment by jlowdermilk** [comment]
It looks like the beta component actually is installed for shared gcloud. Maybe someone recently bumped it? @k8s-bot retest this please

**Comment by mikedanese** [comment]
Please reassign when this is ready to merge.

**Comment by wojtek-t** [comment]
@jlowdermilk (sorry for late response). It seems that beta is not installed - i.e. after rebasing today it's still not there. Is there any way to work around it? Is it possible to proceed with that change?

**Comment by jlowdermilk** [comment]
@wojtek-t, sorry I was looking at the wrong project. I don't have access to the pr builder jenkins master to install the beta component, but I've pinged @ixdy, and will update it shortly.

**Comment by jlowdermilk** [comment]
@k8s-bot test this please

**Comment by wojtek-t** [comment]
>
@wojtek-t, sorry I was looking at the wrong project. I don't have access to the pr builder jenkins master to install the beta component, but I've pinged @ixdy, and will update it shortly.

@ixdy - did you have chance to look into it?

**Comment by wojtek-t** [comment]
This is now ready for review.

**Comment by wojtek-t** [comment]
@roberthbailey - can you please take a look (or delegate)?

File: cluster/gce/util.sh [hunk 2]
```diff
@@ -695,16 +677,16 @@ function kube-up {
   write-node-env
   create-node-instance-template
 
-  gcloud preview managed-instance-groups --zone "${ZONE}" \
+  gcloud beta compute instance-groups managed \
```
Comment (roberthbailey) [inline cluster/gce/util.sh]: When I tested a similar command yesterday this was working for me without the 'beta' cmd group, so I think we can remove that from the script.

**Comment by roberthbailey** [comment]
LGTM, but I think we should use `gcloud compute instance-groups` rather than `gcloud beta compute instance-groups` in all of these commands.

File: cluster/gce/util.sh [hunk 3]
```diff
@@ -695,16 +677,16 @@ function kube-up {
...
       --project "${PROJECT}" \
+      --zone "${ZONE}" \
       --base-instance-name "${NODE_INSTANCE_PREFIX}" \
       --size "${NUM_MINIONS}" \
       --template "${NODE_INSTANCE_PREFIX}-template" || true;
-  # TODO: this should be true when the above create managed-instance-group
-  # command returns, but currently it returns before the instances come up due
-  # to gcloud's deficiency.
-  wait-for-minions-to-run
+  gcloud beta compute instance-groups managed wait-until-stable \
```
Comment (ixdy) [inline cluster/gce/util.sh]: probably need `--project` here too

File: cluster/gce/util.sh (hunk 2 above)
Comment (wojtek-t) [inline cluster/gce/util.sh]: Yes - it seems to work now.

Done.

File: cluster/gce/util.sh (hunk 3 above)
Comment (wojtek-t) [inline cluster/gce/util.sh]: That's not needed, but I've added it for consistency.

**Comment by wojtek-t** [comment]
@roberthbailey - PTAL

**Comment by roberthbailey** [comment]
LGTM.

**Comment by roberthbailey** [comment]
@wojtek-t Please cherry pick this into the 1.0 release branch (after it's been merged) so that our release branch validation tests are using the current API for MIGs as well (I doubt the preview commands will work for much longer).

**Comment by brendandburns** [comment]
@k8s-bot test this [testing build queue, sorry for the noise]

**Comment by brendandburns** [comment]
Automatic merge from SubmitQueue

**Pull request merged**